vcap_utils/vcap_utils/backends/backend_rpc_process.py

```diff
@@ -0,0 +1,204 @@
+import multiprocessing
+import threading
+import queue
+import gc
+from typing import Type, List, Any, Dict, Tuple, Any, NoReturn
+from concurrent.futures import Future
+from uuid import uuid4, UUID
+from threading import RLock
+from typing import NamedTuple
+
+import numpy as np
+
+from vcap import BaseBackend, DETECTION_NODE_TYPE
+
+
+class RpcRequest(NamedTuple):
+    request_id: str
+    """A string UUID"""
+    function: str
+    """The function to run"""
+    args: Tuple[Any]
+    """Arguments for the function"""
+    kwargs: Dict[str, Any]
+    """Keyword arguments for the function"""
+
+
+class RpcResponse(NamedTuple):
+    request_id: str
+    """A string UUID corresponding to the RpcRequest"""
+    result: Any
+    """The result of the function that was run"""
+    exception: BaseException
+    """If the function threw an exception, this will hold the exception. 
+    In the case when the function failed with an exceptio, RpcResponse.result
+    will be None."""
+
+
+def _rpc_server(
+        incoming: multiprocessing.Queue,
+        outgoing: multiprocessing.Queue,
+        shutdown: multiprocessing.Event,
+        num_workers: int,
+        backend_class, args, kwargs):
+    """ An RPC server for running Backend functions.
+    :param incoming: Receives RpcRequests through here
+    :param outgoing: Returns RpcResponses here
+    :param shutdown: When set, this signals that the server should shut down.
+    :param num_workers: How many internal threads should exist for processing
+    incoming RpcRequests.
+    :param backend_class: The backend to initialize
+    :param args: *args for the backend_class __init__
+    :param kwargs: *kwargs for the backend_class __init__
+    """
+    # Initialize the backend and catch any errors during initialization
+    try:
+        backend = backend_class(*args, **kwargs)
+        outgoing.put(None)
+    except BaseException as e:
+        outgoing.put(e)
+        return
+
+    pool = multiprocessing.pool.ThreadPool(processes=num_workers)
+
+    def handle_request(request: RpcRequest):
+        try:
+            result = getattr(backend, request.function)(  # noqa: F821
```

Comment:
Can this be split into two statements for readability?

Reply:
Sure. I'm going to keep both statements in the same try/catch since I would want to catch any exceptions from either statement and send it back through the pipeline anyways.

Reply:
for sure